src/System.Memory/ref/System.Memory.cs

```diff
@@ -112,6 +112,21 @@ public static partial class MemoryExtensions
 }
 namespace System.Buffers
 {
+    public sealed partial class ArrayBufferWriter<T> : System.Buffers.IBufferWriter<T>, System.IDisposable
+    {
+        public ArrayBufferWriter() { }
+        public ArrayBufferWriter(int initialCapacity) { }
+        public int AvailableSpace { get { throw null; } }
+        public int Capacity { get { throw null; } }
+        public int CurrentIndex { get { throw null; } }
+        public System.ReadOnlyMemory<T> OutputAsMemory { get { throw null; } }
+        public System.ReadOnlySpan<T> OutputAsSpan { get { throw null; } }
+        public void Advance(int count) { }
+        public void Clear() { }
+        public void Dispose() { }
+        public System.Memory<T> GetMemory(int sizeHint = 0) { throw null; }
+        public System.Span<T> GetSpan(int sizeHint = 0) { throw null; }
```

Comment:
We should probably have another API review discussion about these names.

Thank you for making these implicitly implemented rather than explicitly.  The crux of my feedback was that, without doing so or exposing similar functionality, the type was useless without being cast to the interface.  The problem now, though, is that these names are a bit confusing, especially with OutputAsMemory and OutputAsSpan exposed on the type.  We could still use explicit interface implementations if we wanted to and then expose this functionality as public methods with different names, in order to make them less confusing when using the public surface area of the type directly rather than via the interface.  Or, we could leave them as is, and either accept the small confusion or potentially come up with different names for OutputAsMemory/Span.